src/NLog/Layouts/XmlLayout.cs

```diff
@@ -0,0 +1,504 @@
+        /// </summary>
+        /// <docgen category='LogEvent Properties XML Options' order='10' />
+        public bool IncludeMdlc { get; set; }
+#endif
+
+        /// <summary>
+        /// Gets or sets the option to include all properties from the log event (as XML)
+        /// </summary>
+        /// <docgen category='LogEvent Properties XML Options' order='10' />
+        public bool IncludeAllProperties { get; set; }
+
+        /// <summary>
+        /// List of property names to exclude when <see cref="IncludeAllProperties"/> is true
+        /// </summary>
+        /// <docgen category='LogEvent Properties XML Options' order='10' />
+#if NET3_5
+        public HashSet<string> ExcludeProperties { get; set; }
+#else
+        public ISet<string> ExcludeProperties { get; set; }
+#endif
+
+        /// <summary>
+        /// XML tag name to use when rendering properties
+        /// </summary>
+        /// <remarks>
+        /// Support string-format where {0} means property-key-name
+        /// 
+        /// Skips closing element tag when having configured <see cref="PropertiesNodeValueAttribute"/>
+        /// </remarks>
+        /// <docgen category='LogEvent Properties XML Options' order='10' />
+        public string PropertiesNodeName { get { return _propertiesNodeName; } set { _propertiesNodeName = value; _propertiesNodeNameFormat = value?.IndexOf('{') >= 0; } }
+        private string _propertiesNodeName = "property";
+        private bool _propertiesNodeNameFormat;
+
+        /// <summary>
+        /// XML attribute format to use when rendering property-key
+        /// </summary>
+        /// <remarks>
+        /// Replaces newlines with &#13;&#10;
+        /// </remarks>
+        /// <docgen category='LogEvent Properties XML Options' order='10' />
+        public string PropertiesNodeKeyAttribute { get; set; } = "key";
+
+        /// <summary>
+        /// XML attribute format to use when rendering property-value
+        /// 
+        /// When null (or empty) then value is formatted as XML-element-value
+        /// </summary>
+        /// <remarks>
+        /// Replaces newlines with &#13;&#10;
+        /// 
+        /// Skips closing element tag when using attribute for value
+        /// </remarks>
+        /// <docgen category='LogEvent Properties XML Options' order='10' />
+        public string PropertiesNodeValueAttribute { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlLayout"/> class.
+        /// </summary>
+        public XmlLayout()
+            :this("logevent", null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlLayout"/> class.
+        /// </summary>
+        /// <param name="nodeName">The name of the top XML node</param>
+        /// <param name="nodeValue">The value of the top XML node</param>
+        public XmlLayout(string nodeName, Layout nodeValue)
```

Comment:
please move the ctors above the properties, thx!

Reply:
Changed